[Bug] don't sign with --deep on MacOS

### Description of the issue

Hello there, I come from over at nixpkgs. I've been trying to provide a build of ryujinx for MacOS + Nix users. This has been a bit of a journey, and what I noticed is that it's failing at codesigning postBuild.

I would like to hint towards --deep being deprecated / removed and [why it shouldn't be used.](https://developer.apple.com/forums/thread/129980)

Here's a concrete log of what's ocurring:

```nix
*-- snip --*
==  Ryujinx.Headless.SDL2 -> /private/tmp/nix-build-ryujinx-1.2.72.drv-2/source/src/Ryujinx.Headless.SDL2/bin/Release/net8.0/Ryujinx.Headless.SDL2.dll
  The following argument was not expected: --deep
  Run with --help for more information.
=/private/tmp/nix-build-ryujinx-1.2.72.drv-2/source/src/Ryujinx.Headless.SDL2/Ryujinx.Headless.SDL2.csproj(20,5): error MSB3073: The command "codesign --entitlements '/private/tmp/nix-build-ryujinx-1.2.72.drv-2/source/src/Ryujinx.Headless.SDL2/../../distribution/macos/entitlements.xml' -f --deep -s - '/private/tmp/nix-build-ryujinx-1.2.72.drv-2/source/src/Ryujinx.Headless.SDL2/bin/Release/net8.0/Ryujinx.Headless.SDL2'" exited with code 109.
  Ryujinx.Graphics.Nvdec.FFmpeg -> /private/tmp/nix-build-ryujinx-1.2.72.drv-2/source/src/Ryujinx.Graphics.Nvdec.FFmpeg/bin/Release/net8.0/Ryujinx.Graphics.Nvdec.FFmpeg.dll
  Ryujinx.Audio.Backends.OpenAL -> /private/tmp/nix-build-ryujinx-1.2.72.drv-2/source/src/Ryujinx.Audio.Backends.OpenAL/bin/Release/net8.0/Ryujinx.Audio.Backends.OpenAL.dll
  Ryujinx.UI.Common -> /private/tmp/nix-build-ryujinx-1.2.72.drv-2/source/src/Ryujinx.UI.Common/bin/Release/net8.0/Ryujinx.UI.Common.dll
  Ryujinx.UI.LocaleGenerator -> /private/tmp/nix-build-ryujinx-1.2.72.drv-2/source/src/Ryujinx.UI.LocaleGenerator/bin/Release/netstandard2.0/Ryujinx.UI.LocaleGenerator.dll
==  Ryujinx -> /private/tmp/nix-build-ryujinx-1.2.72.drv-2/source/src/Ryujinx/bin/Release/net8.0/Ryujinx.dll
  The following argument was not expected: --deep
  Run with --help for more information.
=/private/tmp/nix-build-ryujinx-1.2.72.drv-2/source/src/Ryujinx/Ryujinx.csproj(17,5): error MSB3073: The command "codesign --entitlements '/private/tmp/nix-build-ryujinx-1.2.72.drv-2/source/src/Ryujinx/../../distribution/macos/entitlements.xml' -f --deep -s - '/private/tmp/nix-build-ryujinx-1.2.72.drv-2/source/src/Ryujinx/bin/Release/net8.0/Ryujinx'" exited with code 109.
  Ryujinx.UI.Common -> /private/tmp/nix-build-ryujinx-1.2.72.drv-2/source/src/Ryujinx.UI.Common/bin/Release/net8.0/Ryujinx.UI.Common.dll
  Ryujinx.Horizon.Generators -> /private/tmp/nix-build-ryujinx-1.2.72.drv-2/source/src/Ryujinx.Horizon.Generators/bin/Release/netstandard2.0/Ryujinx.Horizon.Generators.dll
  Ryujinx.Graphics.Vulkan -> /private/tmp/nix-build-ryujinx-1.2.72.drv-2/source/src/Ryujinx.Graphics.Vulkan/bin/Release/net8.0/Ryujinx.Graphics.Vulkan.dll
  Spv.Generator -> /private/tmp/nix-build-ryujinx-1.2.72.drv-2/source/src/Spv.Generator/bin/Release/net8.0/Spv.Generator.dll
  Ryujinx.UI.LocaleGenerator -> /private/tmp/nix-build-ryujinx-1.2.72.drv-2/source/src/Ryujinx.UI.LocaleGenerator/bin/Release/netstandard2.0/Ryujinx.UI.LocaleGenerator.dll
  Ryujinx.Horizon.Common -> /private/tmp/nix-build-ryujinx-1.2.72.drv-2/source/src/Ryujinx.Horizon.Common/bin/Release/net8.0/Ryujinx.Horizon.Common.dll
  Ryujinx.Horizon -> /private/tmp/nix-build-ryujinx-1.2.72.drv-2/source/src/Ryujinx.Horizon/bin/Release/net8.0/Ryujinx.Horizon.dll
  Ryujinx.Horizon.Kernel.Generators -> /private/tmp/nix-build-ryujinx-1.2.72.drv-2/source/src/Ryujinx.Horizon.Kernel.Generators/bin/Release/netstandard2.0/Ryujinx.Horizon.Kernel.Generators.dll
  Ryujinx.HLE.Generators -> /private/tmp/nix-build-ryujinx-1.2.72.drv-2/source/src/Ryujinx.HLE.Generators/bin/Release/netstandard2.0/Ryujinx.HLE.Generators.dll

Build FAILED.

/private/tmp/nix-build-ryujinx-1.2.72.drv-2/source/src/Ryujinx.HLE/obj/Release/net8.0/Ryujinx.HLE.Generators/Ryujinx.HLE.Generators.IpcServiceGenerator/IUserInterface.g.cs(6,84): warning CS8669: The annotation for nullable reference types should only be used in code within a '#nullable' annotations context. Auto-generated code requires an explicit '#nullable' directive in source. [/private/tmp/nix-build-ryujinx-1.2.72.drv-2/source/src/Ryujinx.HLE/Ryujinx.HLE.csproj]
/private/tmp/nix-build-ryujinx-1.2.72.drv-2/source/src/Ryujinx.HLE/obj/Release/net8.0/Ryujinx.HLE.Generators/Ryujinx.HLE.Generators.IpcServiceGenerator/IUserInterface.g.cs(6,26): warning CS8669: The annotation for nullable reference types should only be used in code within a '#nullable' annotations context. Auto-generated code requires an explicit '#nullable' directive in source. [/private/tmp/nix-build-ryujinx-1.2.72.drv-2/source/src/Ryujinx.HLE/Ryujinx.HLE.csproj]
/private/tmp/nix-build-ryujinx-1.2.72.drv-2/source/src/Ryujinx.Headless.SDL2/Ryujinx.Headless.SDL2.csproj(20,5): error MSB3073: The command "codesign --entitlements '/private/tmp/nix-build-ryujinx-1.2.72.drv-2/source/src/Ryujinx.Headless.SDL2/../../distribution/macos/entitlements.xml' -f --deep -s - '/private/tmp/nix-build-ryujinx-1.2.72.drv-2/source/src/Ryujinx.Headless.SDL2/bin/Release/net8.0/Ryujinx.Headless.SDL2'" exited with code 109.
/private/tmp/nix-build-ryujinx-1.2.72.drv-2/source/src/Ryujinx/Ryujinx.csproj(17,5): error MSB3073: The command "codesign --entitlements '/private/tmp/nix-build-ryujinx-1.2.72.drv-2/source/src/Ryujinx/../../distribution/macos/entitlements.xml' -f --deep -s - '/private/tmp/nix-build-ryujinx-1.2.72.drv-2/source/src/Ryujinx/bin/Release/net8.0/Ryujinx'" exited with code 109.
```

### Reproduction steps

I currently build a custom derivation of https://github.com/NixOS/nixpkgs/blob/nixos-24.05/pkgs/by-name/ry/ryujinx/package.nix.

I can provide the source but I don't think it's exactly necessary since the issue is with codesigning.

Using --deep in the postBuild step doesn't work. Similarly when removing it, the status codesign returns is 134. The relevant log out is this:

```nix
==  Ryujinx -> /private/tmp/nix-build-ryujinx-1.2.72.drv-2/source/src/Ryujinx/bin/Release/net8.0/Ryujinx.dll
  libc++abi: terminating due to uncaught exception of type std::runtime_error: Failed to spawn codesign_allocate: No such file or directory
=/private/tmp/nix-build-ryujinx-1.2.72.drv-2/source/src/Ryujinx/Ryujinx.csproj(17,5): error MSB3073: The command "codesign --entitlements '/private/tmp/nix-build-ryujinx-1.2.72.drv-2/source/src/Ryujinx/../../distribution/macos/entitlements.xml' -f -s - '/private/tmp/nix-build-ryujinx-1.2.72.drv-2/source/src/Ryujinx/bin/Release/net8.0/Ryujinx'" exited with code 134.
```

I am not sure how to handle this as I assume it's an upstream issue (from a nixpkgs standpoint), so that's why I'm reporting it here.

### Log file

N/A

### OS

OSX M3 aarch64-apple-darwin

### Ryujinx version

1.2.72

### Game version

_No response_

### CPU

_No response_

### GPU

_No response_

### RAM

_No response_

### List of applied mods

_No response_

### Additional context?

_No response_

--- a/distribution/macos/create_app_bundle.sh
+++ b/distribution/macos/create_app_bundle.sh
@@ -46,5 +46,5 @@ then
     rcodesign sign --entitlements-xml-path "$ENTITLEMENTS_FILE_PATH" "$APP_BUNDLE_DIRECTORY"
 else
     echo "Usign codesign for ad-hoc signing"
-    codesign --entitlements "$ENTITLEMENTS_FILE_PATH" -f --deep -s - "$APP_BUNDLE_DIRECTORY"
+    codesign --entitlements "$ENTITLEMENTS_FILE_PATH" -f -s - "$APP_BUNDLE_DIRECTORY"
 fi

--- a/distribution/macos/create_macos_build_ava.sh
+++ b/distribution/macos/create_macos_build_ava.sh
@@ -99,7 +99,7 @@ then
     rcodesign sign --entitlements-xml-path "$ENTITLEMENTS_FILE_PATH" "$UNIVERSAL_APP_BUNDLE"
 else
     echo "Using codesign for ad-hoc signing"
-    codesign --entitlements "$ENTITLEMENTS_FILE_PATH" -f --deep -s - "$UNIVERSAL_APP_BUNDLE"
+    codesign --entitlements "$ENTITLEMENTS_FILE_PATH" -f -s - "$UNIVERSAL_APP_BUNDLE"
 fi
 
 echo "Creating archive"

--- a/distribution/macos/create_macos_build_headless.sh
+++ b/distribution/macos/create_macos_build_headless.sh
@@ -95,7 +95,7 @@ else
     echo "Using codesign for ad-hoc signing"
     for FILE in "$UNIVERSAL_OUTPUT"/*; do
         if [[ $(file "$FILE") == *"Mach-O"* ]]; then
-            codesign --entitlements "$ENTITLEMENTS_FILE_PATH" -f --deep -s - "$FILE"
+            codesign --entitlements "$ENTITLEMENTS_FILE_PATH" -f -s - "$FILE"
         fi
     done    
 fi

--- a/src/Ryujinx.Headless.SDL2/Ryujinx.Headless.SDL2.csproj
+++ b/src/Ryujinx.Headless.SDL2/Ryujinx.Headless.SDL2.csproj
@@ -1,4 +1,4 @@
-﻿<Project Sdk="Microsoft.NET.Sdk">
+<Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
     <TargetFramework>net8.0</TargetFramework>
@@ -17,7 +17,7 @@
   </ItemGroup>
 
   <Target Name="PostBuild" AfterTargets="PostBuildEvent" Condition="$([MSBuild]::IsOSPlatform('OSX'))">
-    <Exec Command="codesign --entitlements '$(ProjectDir)..\..\distribution\macos\entitlements.xml' -f --deep -s $(SigningCertificate) '$(TargetDir)$(TargetName)'" />
+    <Exec Command="codesign --entitlements '$(ProjectDir)..\..\distribution\macos\entitlements.xml' -f -s $(SigningCertificate) '$(TargetDir)$(TargetName)'" />
   </Target>
 
   <ItemGroup>

--- a/src/Ryujinx/Ryujinx.csproj
+++ b/src/Ryujinx/Ryujinx.csproj
@@ -14,7 +14,7 @@
   </PropertyGroup>
 
   <Target Name="PostBuild" AfterTargets="PostBuildEvent" Condition="$([MSBuild]::IsOSPlatform('OSX'))">
-    <Exec Command="codesign --entitlements '$(ProjectDir)..\..\distribution\macos\entitlements.xml' -f --deep -s $(SigningCertificate) '$(TargetDir)$(TargetName)'" />
+    <Exec Command="codesign --entitlements '$(ProjectDir)..\..\distribution\macos\entitlements.xml' -f -s $(SigningCertificate) '$(TargetDir)$(TargetName)'" />
   </Target>
 
   <PropertyGroup Condition="'$(RuntimeIdentifier)' != ''">

